types_test.go

```diff
@@ -129,3 +130,37 @@ func TestToSignedTreeHead(t *testing.T) {
 		})
 	}
 }
+
+// TestSignedCertificateTimestampMarshal tests that a
+// SignedCertificateTimestamp object can be marshaled to JSON and produces the
+// expected JSON object.
+func TestSignedCertificateTimestampMarshal(t *testing.T) {
+	// Define a dummy SCT object with some fake information
+	sctObj := &SignedCertificateTimestamp{
+		SCTVersion: V1,
+		LogID: LogID{
+			KeyID: [32]byte{0xC0, 0xFF, 0xEE},
+		},
+		Timestamp:  uint64(12345678),
+		Extensions: CTExtensions{0x13, 0x37},
+		Signature: DigitallySigned{
+			Algorithm: tls.SignatureAndHashAlgorithm{
+				Hash:      tls.SHA256,
+				Signature: tls.ECDSA,
+			},
+		},
+	}
+
+	// The SCT object should marshal to JSON without any error
+	jsonBytes, err := json.Marshal(sctObj)
+	if err != nil {
+		t.Errorf("Failed to marshal SignedCertificateTransparency ob to JSON: %s", err.Error())
+	}
+
+	// The SCT object should marshal to the expected JSON
+	expectedJSON := `{"sct_version":0,"id":{"KeyID":[192,255,238,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]},"timestamp":12345678,"extensions":"Ezc=","Signature":"BAMAAA=="}`
```

Comment:
This doesn't look right -- I'd expect something more like `{"sct_version":0,"id":"wP/uAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=","timestamp":12345678,"extensions":"","signature":"BAMAAA=="}`

I think the problem is that `LogID` and `CTExtensions` are `struct` types (IIRC they're set up that way so the TLS-encoding of SCTs works, which is needed for embedded SCTs), so the JSON encoding puts a dictionary in there rather than just the base64 data.

I guess it would be possible to get around that by adding an [`SignedCertificateTimestamp.MarshalJSON()`](https://golang.org/pkg/encoding/json/#Marshaler) method, but it might be easier just to stick to having two types (`SignedCertificateTimestamp` when TLS-encoding is needed, `AddChainResponse` for JSON-encoded responses).

Reply:
Hmm, good catch!

> I guess it would be possible to get around that by adding an SignedCertificateTimestamp.MarshalJSON() method, but it might be easier just to stick to having two types (SignedCertificateTimestamp when TLS-encoding is needed, AddChainResponse for JSON-encoded responses).

I think you're right. I'm happy to close this PR and the associated issue knowing that `AddChainResponse` can serve my purposes. It seems like a more direct solution than the additional complexity of having to implement the JSON marshalling explicitly.

Thanks for the review/input @daviddrysdale